manager/director/apps/sites/forms.py

```diff
@@ -161,6 +171,35 @@ def clean(self) -> Dict[str, Any]:
             if "domain" in cleaned_data and cleaned_data["domain"].endswith("tjhsst.edu"):
                 self.add_error("domain", "Only administrators can add tjhsst.edu domains")
 
+        # Check if this domain CNAMEs to the proper value
+        if "domain" in cleaned_data and cleaned_data["domain"] and not self.user_is_superuser:
+            try:
+                # Attempt to resolve this domain (check for CNAME records)
+                result = dns.resolver.resolve(
+                    cleaned_data["domain"], "CNAME", raise_on_no_answer=False
+                )
+                # If the result set is None (no CNAMEs) or if there are no records that
+                # point to the CNAME configured in settings, raise a ValueError (which is caught)
+                if result.rrset is None or not any(
+                    [
+                        settings.DIRECTOR_CUSTOM_DOMAIN_FQDN_CNAME in response.to_text()
```

Comment:
This validation seems a little loose (for example, somebody could set `director.csl.tjhsst.eduXXX` and it would be accepted). I'm guessing you did this to handle the trailing dot in CNAME records; couldn't that be done with `.rstrip(".")`?